src/Subscribers.php

```diff
@@ -108,22 +109,35 @@
             echo json_encode(array('status' => 'error', 'result' => 'List not found'));
             $app->stop();
         }
-
-        $sth = $db->prepare('SELECT * FROM subscribers WHERE list = :list AND email = :email');
-        $sth->execute(array('list' => $post['list'], 'email' => $post['email']));
-
-        if(count($sth->fetchAll(PDO::FETCH_ASSOC)) <= 0)
+        
+        // Unsubscribe from all lists
+        if($list['unsubscribe_all_list'] === '1')
         {
-            echo json_encode(array('status' => 'error', 'result' => 'User not on the list'));
-            $app->stop();
+            //check if user exists in db
+            $sth = $db->prepare('SELECT * FROM subscribers WHERE email = :email');
+            $sth->execute(array('email' => $post['email']));
+            $subscribers = $sth->fetchAll(PDO::FETCH_ASSOC);
+            foreach ($subscribers as $subscriber) 
```

Comment:
you don't have to iterate subscribers here, you can do just one update:
```UPDATE subscribers SET `unsubscribed` = "1", `timestamp` = :timestamp WHERE email = :email```
it will unsubscribe user from all lists